### PR TITLE
Adding CDI support, as well as MicroProfile Config support.

### LINKED ISSUE #3: Integration with CDI
Explain how to bootstrap a client that is just an interface into a CDI bean, without requiring custom producers.

### PR DESCRIPTION
**Pull request by johnament**
This closes #3 & #4 

I went ahead and combined these two tickets.  There were two reasons for it:

- I didn't want to introduce a hard coded way to have a URL set, it should always be configured.
- Most of the configuration defined is optional, since they can be defined directly on the interface.

Any thoughts on this approach @andymc12 or @reta ?

### CODE REVIEW

File: spec/src/main/asciidoc/cdi.asciidoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @Path("/greet")
+    Response greet();
+}
+----
+
+The following properties will then be available:
+
+- `com.mycompany.remoteServices.MyServiceClient.url`: The base URL to use for this service, the equivalent of the `baseUrl` method.  This property is considered required, however implementations may have other ways to define these URLs.
+- `com.mycompany.remoteServices.MyServiceClient.scope`: The fully qualified classname to a CDI scope to use for injection, defaults to `javax.enterprise.context.Dependent` as mentioned above.
+- `com.mycompany.remoteServices.MyServiceClient.providers`: A comma separated list of providers to include in the client, the equivalent of the `register` method or the `@RegisterProviders` annotation.
```
Comment (reta) [inline spec/src/main/asciidoc/cdi.asciidoc]: I think it is implied, but may be better to be explicit about features as well, for example, `.. list of providers and features ...`

Comment (johnament) [reply spec/src/main/asciidoc/cdi.asciidoc]: ack'd, will include in the next update.

File: spec/src/main/asciidoc/cdi.asciidoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+package com.mycompany.remoteServices;
+
+public interface MyServiceClient {
+    @GET
+    @Path("/greet")
+    Response greet();
+}
+----
+
+The following properties will then be available:
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/cdi.asciidoc]: Do you mean the MyServiceClient provides the following properties? It will be a config source. If so, what is the ordinal for this config source, as these properties might be overridden by other config sources? The ordinal should be configurable as well.

File: spec/src/main/asciidoc/cdi.asciidoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- `com.mycompany.remoteServices.MyServiceClient.url`: The base URL to use for this service, the equivalent of the `baseUrl` method.  This property is considered required, however implementations may have other ways to define these URLs.
```
Comment (johnament) [inline spec/src/main/asciidoc/cdi.asciidoc]: Based on call discussion, we're going to use a suffix based approach for the properties.  So they will instead look like `com.mycompany.remoteServices.MyServiceClient/mp-rest/url` and likewise for `providers` and `scope`

File: spec/src/main/asciidoc/cdi.asciidoc (hunk 2 above)
Comment (johnament) [reply spec/src/main/asciidoc/cdi.asciidoc]: Based on the discussion on the call, will rephrase this to be `The values of the following properties will be provided via MicroProfile Config`:

File: spec/src/main/asciidoc/cdi.asciidoc (hunk 1 above)
Comment (johnament) [reply spec/src/main/asciidoc/cdi.asciidoc]: We're going to create a follow up ticket for JAX-RS Features.

File: spec/src/main/asciidoc/cdi.asciidoc (hunk 3 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/cdi.asciidoc]: It fits better with the naming convention of Fault Tolerance config properties.

**Comment by johnament** [comment]
I was thinking about this a bit more.  I think there's a bit more, wanted to put this out there.  Should we require a qualifier on the injection point, to indicate that this is meant to be the remote client?  If we don't, it becomes difficult to differentiate what is meant to be the rest client and other use cases.

**Comment by johnament** [comment]
We'll add a new qualifier, name TBD, that represents that this interface should be a rest client.  The qualifier should not be required on the injection point, only if there are both client and server use cases.
